cli/admin/src/main/java/io/pravega/cli/admin/dataRecovery/DurableDataLogRepairCommand.java

```diff
@@ -0,0 +1,914 @@
+        boolean sealed = getBooleanUserInput("Is the Segment sealed? [true/false]: ");
+        boolean sealedInStorage = getBooleanUserInput("Is the Segment sealed in storage? [true/false]: ");
+        boolean deleted = getBooleanUserInput("Is the Segment deleted? [true/false]: ");
+        boolean deletedInStorage = getBooleanUserInput("Is the Segment deleted in storage? [true/false]: ");
+        output("You are about to start adding Attributes to the SegmentProperties instance.");
+        boolean finishInputCommands = !confirmContinue();
+        Map<AttributeId, Long> attributes = new HashMap<>();
+        while (!finishInputCommands) {
+            output("Creating an AttributeUpdateCollection for this operation.");
+            try {
+                AttributeId attributeId = AttributeId.fromUUID(UUID.fromString(getStringUserInput("Introduce UUID for this Attribute: ")));
+                long value = getLongUserInput("Introduce the Value for this Attribute:");
+                attributes.put(attributeId, value);
+            } catch (NumberFormatException ex) {
+                outputError("Wrong input argument.");
+                ex.printStackTrace();
+            } catch (Exception ex) {
+                outputError("Some problem has happened.");
+                ex.printStackTrace();
+            }
+            output("You can continue adding AttributeUpdates to the AttributeUpdateCollection.");
+            finishInputCommands = !confirmContinue();
+        }
+        long lastModified = getLongUserInput("Introduce last modified timestamp for the Segment (milliseconds): ");
+        return StreamSegmentInformation.builder().name(segmentName).startOffset(offset).length(length).storageLength(storageLength)
+                .sealed(sealed).deleted(deleted).sealedInStorage(sealedInStorage).deletedInStorage(deletedInStorage)
+                .attributes(attributes).lastModified(new ImmutableDate(lastModified)).build();
+    }
+
+    /**
+     * Method to create a {@link AttributeUpdateCollection} object to fill the {@link Operation}s that require it.
+     *
+     * @return New {@link AttributeUpdateCollection} object with user-defined content.
+     */
+    @VisibleForTesting
+    AttributeUpdateCollection createAttributeUpdateCollection() {
+        AttributeUpdateCollection attributeUpdates = new AttributeUpdateCollection();
+        output("You are about to start adding AttributeUpdates to the AttributeUpdateCollection.");
+        boolean finishInputCommands = !confirmContinue();
+        while (!finishInputCommands) {
+            output("Creating an AttributeUpdateCollection for this operation.");
+            try {
+                AttributeId attributeId = AttributeId.fromUUID(UUID.fromString(getStringUserInput("Introduce UUID for this AttributeUpdate: ")));
+                AttributeUpdateType type = AttributeUpdateType.get((byte) getIntUserInput("Introduce AttributeUpdateType for this AttributeUpdate" +
```

Comment:
FYI once serialized in the log, the update type and comparison value REALLY do not matter anymore. They are pre-validated initially when we get the operation but later on the value has already been precomputed and accepted. So I think we can simplify this by removing "AttributeUpdateType" and "Comparison value". Just use `Replace`.

Reply:
Yes, but if someone looks at an `AttributeUpdate` to understand what it is, he/she may be confused about just adding some fields and not others. Even though they may not be necessary, the current approach may be more directly understandable from a non-expert person in the Segment Store.